src/analyses/memLeak.ml

```diff
@@ -20,6 +20,105 @@ struct
   let context _ d = d
 
   (* HELPER FUNCTIONS *)
+  let get_global_vars () =
+    (* Filtering by GVar seems to account for declarations, as well as definitions of global vars *)
+    List.filter_map (function GVar (v, _, _) -> Some v | _ -> None) !Cilfacade.current_file.globals
+
+  let get_global_struct_ptr_vars () =
+    List.filter_map (function GVar (v, _, _) -> Some v | _ -> None) !Cilfacade.current_file.globals
+    |> List.filter (fun v -> match v.vtype with TPtr (TComp _, _) | TPtr ((TNamed ({ttype = TComp _; _}, _)), _) -> true | _ -> false)
+
+  let get_global_struct_non_ptr_vars () =
+    List.filter_map (function GVar (v, _, _) -> Some v | _ -> None) !Cilfacade.current_file.globals
+    |> List.filter (fun v -> match v.vtype with TComp (_, _) | (TNamed ({ttype = TComp _; _}, _)) -> true | _ -> false)
```

Comment:
Wouldn't `List.partition` give both in a single pass?

Reply:
I'll try it out, thanks for the tip!

Reply:
I'm only not sure about the predicate formulation for `List.partition`, namely: we'd have the `true` case for the `TPtr` structs and the `false` one for the structs without a `TPtr` type. In this case what should we do with the rest of the types (otherwise, the match wouldn't be complete)?

Reply:
Yes, then it is not quite a good usecase for `partition`. We could probably come up with an efficient thing that splits it into two lists and discards the rest in linear time, but I don't think this is crucial for this year's sv-comp.